proto/kvrpcpb.proto

```diff
@@ -120,6 +128,9 @@ message PrewriteRequest {
     uint64 max_commit_ts = 14;
     // The level of assertion to use on this prewrte request.
     AssertionLevel assertion_level = 15;
+    // for_update_ts constriants that should be checked when prewriting a pessimistic transaction.
+    // See https://github.com/tikv/tikv/issues/14311
+    repeated ForUpdateTSConstraint for_update_ts_constraints = 16;
```

Comment:
As the fair lock mode supports only single key lock now, does it mean for a transaction the maximum length of this array is `txn.total_statment_number`?

Reply:
No. It's possible that a statement calls `LockKeys` more than once and locks one key each.